Create a landing page for the Hosting Team (#3142)
jenkins-infra/jenkins.io: content/project/teams/hosting.adoc

---
layout: project
title: "Hosting Team"
section: projects
tags:
- hosting
- plugin-site
links:
  mailinglist: "/mailing-lists/#jenkinsci-dev-googlegroups-com"
---

> Jenkins Hosting Team handles various requests related to hosting of open-source plugins and other components. (forking and transferring plugins, managing permissions and update centers, etc.).

Reviewer: ```suggestion
The Jenkins Hosting Team handles various requests related to hosting of open-source plugins and other components. (forking and transferring plugins, managing permissions and update centers, etc.).
```

> The Jenkins Hosting Team handles various requests related to hosting of open-source plugins and other components. (forking and transferring plugins, managing permissions and update centers, etc.).

Reviewer 2: Minor punctuation change
```suggestion
The Jenkins Hosting Team handles various requests related to hosting of open-source plugins and other components (forking and transferring plugins, managing permissions and update centers, etc.).
```

> There are multiple contributors involved in this activity on regular basis.

Reviewer: ```suggestion
There are multiple contributors involved in this activity on a regular basis.
```

Author: Apparently it was spelling which suggested removing the article there :) I also think this is more correct

> == Team responsibilities

Reviewer: ```suggestion
== Team Responsibilities
```

> * Triage and processing of new plugin HOSTING requests in Jenkins Jira.

Reviewer: Maybe put a link to the HOSTING project in Jira?

Author: ```suggestion
* Triage and processing of new plugin link:https://issues.jenkins-ci.org/projects/HOSTING[HOSTING] requests in Jenkins Jira.
```

>   There is some automation around this process, but human reviews are needed to...

Reviewer: ```suggestion
  There is some automation around this process, but human reviews are needed too...
```

> ** Ensure that link:/doc/developer/publishing/preparation[plugin publishing requirements] are followed:

Reviewer 3: Strange this link gives me 404 :(

> naming conventions and `artifactId`s, documentation, duplication with existing plugins, license compliance, etc.]

Reviewer: ```suggestion
naming conventions and `artifactId`s, `groupId`s, documentation, duplication of existing plugins, license compliance, etc.]
```

> ** Do sanity check of plugins for security issues.

Reviewer: ```suggestion
** Check plugin compliance with security best practices and recommendations.
```

* Managing plugin developer and release permissions (link:/doc/developer/plugin-governance/managing-permissions/[documentation]).
* Processing link:/doc/developer/plugin-governance/adopt-a-plugin/[plugin adoption] requests in the developer mailing list
* Processing blacklisting and plugin tagging/doc URL requests in the Update Centers for non-security reasons.
* Maintaining the link:/doc/developer/publishing[publishing] and link:/doc/developer/plugin-governance/[plugin governance] documentation.

== Key repositories

* link:https://github.com/jenkins-infra/repository-permissions-updater[Repository Permission Updater] - Manages release permissions for all components hosted in Jenkins Artifactory
* link:https://github.com/jenkins-infra/update-center2[Jenkins Update Center] - Jenkins Update Center repository.
  Includes update center generation logic, including plugin blacklists, security labels and documentation metadata.
* link:https://github.com/jenkins-infra/ircbot[Jenkins IRC Bot] - our ChatOps tool

== How to contribute?

=== Assisting with Q&A and request reviews

> A significant part of this team's work is helping newcomer plugin maintainers to get their plugins hosted and published.

Reviewer: ```suggestion
A significant part of this team's work is helping new plugin maintainers to get their plugins hosted and published.
```

Also, we help with permission management when contributors add new developers or adopt plugins.

> There is a lot of channels for these topics, but all of them public and do not require special permission.

Reviewer: ```suggestion
There are a lot of channels for these topics, but all of them public and do not require special permissions to join.
```

> There are a lot of channels for these topics, but all of them public and do not require special permissions to join.

Reviewer 3: ```suggestion
There are a lot of channels for these topics, but all of them are public and do not require special permissions to join.
```

We invite any interested Jenkins contributor to join the team.

How to start contributing?

. Subscribe to the link:/mailing-lists/#jenkinsci-dev-googlegroups-com[Jenkins Developer Mailing list]
. Subscribe to the link:https://issues.jenkins-ci.org/projects/HOSTING[HOSTING project] in Jenkins Jira
. Subscribe to the link:https://github.com/jenkins-infra/repository-permissions-updater/[Repository Permission Updater] repository
. Subscribe to the link:https://github.com/jenkins-infra/update-center2[Jenkins Update Center] repository.

=== Joining the main hosting team

==== Requesting membership

If you are interested to join the Triage or Copy Editors team, 
you can request membership in the link:https://groups.google.com/d/forum/jenkinsci-dev[Jenkins Developer mailing list].
The request will be processed and discussed by the community, and the decision will be made based on consensus among contributors

Eligibility:

> * Membership in the team requires a track of contributions to the hosting process.

Reviewer: ```suggestion
* Membership in the team requires a track record of contributions to the hosting process.
```

  Assisting users with Q&A and request reviews does not require special permissions.
* All applicants should have a signed link:https://github.com/jenkinsci/infra-cla[Contributor License Agreement] before they apply.

==== Configuration

. Subscribe to all channels and repositories listed in the _Assisting with Q&A and request reviews_ section
. Join the link:/chat/[`#jenkins` IRC channel] and link:/chat/#cloaks[obtain a cloak].
  It will be needed for ChatOps.
. Request voice permission (`+vV`) in the link:/chat/[`#jenkins` IRC channel] to get access to the bot commands.

==== Notes

> * Some requests require Manual changes in GitHub repositories for some requests, e.g. plugin renaming.

Reviewer: ```suggestion
* Some requests require manual changes in GitHub repositories for some requests, e.g. plugin renaming.
```
